tools/pythonpkg/tests/fast/arrow/test_arrow_fetch_recordbatch.py

```diff
@@ -106,3 +106,12 @@ def test_record_coverage(self, duckdb_cursor):
         chunk = record_batch_reader.read_all()
         assert(len(chunk) == 2048)
 
+    def test_record_batch_query_error(self):
+        if not can_run:
+            return
+        duckdb_cursor = duckdb.connect()
+        duckdb_cursor.execute("CREATE table t as select 'foo' as a;")
+        query = duckdb_cursor.execute("SELECT cast(a as double) FROM t")
+        record_batch_reader = query.fetch_record_batch(1024)
+        with pytest.raises(OSError, match='Conversion Error'):
```

Comment:
Is this really throwing an `OSError`?

Reply:
Yes, it's raised in `pyarrow`, for example:
```Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow\ipc.pxi", line 682, in pyarrow.lib.RecordBatchReader.read_next_batch
  File "pyarrow\error.pxi", line 115, in pyarrow.lib.check_status
OSError: Query Failed```

Reply:
Huh, interesting. Happy to merge this, though I might have to do some investigation to see if that exception type can be improved